posts/2024-03-18_xportr_0_4_0/xportr_0_4_0.qmd

```diff
@@ -0,0 +1,122 @@
+![](xportr_design_flow.png){fig-align="center" width="500"}
+
+[`xportr`](https://atorus-research.github.io/xportr/) is currently on version 0.4.0 and you can keep up with all of its improvements by following the [`Changelog here`](https://atorus-research.github.io/xportr/news/index.html)
+
+# Why [`xportr`](https://atorus-research.github.io/xportr/)?
+
+Have I already mentioned that [`xportr`](https://atorus-research.github.io/xportr/) is an essential R package for generating `xpt` datasets? This package includes functions to apply metadata information to your CDISC compliant datasets that were created in R, perform checks and provide useful feedbacks. In this blog post, we'll explore some high level capabilities of [`xportr`](https://atorus-research.github.io/xportr/), and its development.
+
+# How does it work?
+
+You too can experience some of the benefits of [`xportr`](https://atorus-research.github.io/xportr/) by taking a [`deep dive`](https://atorus-research.github.io/xportr/articles/deepdive.html) at [`xportr`](https://atorus-research.github.io/xportr/)!
+
+There are data deliverables and supporting documentation needed for a successful submission to Health Authorities, while you can find a walk through on how you can use `xportr` [`here`](https://atorus-research.github.io/xportr/articles/deepdive.html#what-goes-in-a-submission-to-a-health-authority), let's look at a high level example:
+
+```{r,  message = FALSE, echo = FALSE }
+library(xportr)
+library(magrittr)
+```
+
+Looking at an `adsl` and the define metadata information
+```{r,  message = FALSE, warnings = FALSE }
+adsl <- data.frame(
+  Subj = as.character(123, 456, 789),
+  Different = c("a", "b", "c"),
+  Val = c("1", "2", "3"),
+  Param = c("param1", "param2", "param3")
+)
+```
+
+and the define metadata information
+```{r,  message = FALSE, warnings = FALSE}
+metadata <- data.frame(
+  dataset = "test",
+  variable = c("Subj", "Param", "Val", "NotUsed"),
+  type = c("numeric", "character", "numeric", "character"),
+  format = NA,
+  order = c(1, 3, 4, 2)
+)
+```
+
+Users can use the [`xportr_metadata()`](https://atorus-research.github.io/xportr/reference/xportr_metadata.html) function along with [`xportr_type()`](https://atorus-research.github.io/xportr/reference/xportr_type.html) and [`xportr_order()`](https://atorus-research.github.io/xportr/reference/xportr_order.html) to not only apply metadata attributes to your data but also receive feedback on your data based on your spec. Additionally, some of those functions performing valuable checks like:
+
+  * Variable names must start with a letter (not an underscore), be comprised of only uppercase letters (A-Z), numerals (0-9) and be free of non-ASCII characters, symbols, and underscores.
```

Comment:
Could we list out the other functions here.  The xportr_label, xportr_format and xportr_write with the strict checks should get a mention here for sure.

Reply:
oh and xportr_length